src/dune_rules/melange_rules.ml

```diff
@@ -6,9 +6,7 @@ let lib_output_dir ~target_dir ~lib_dir =
     (Path.Build.drop_build_context_exn lib_dir)
 
 let make_js_name ~dst_dir m =
-  let name =
-    Module_name.Unique.artifact_filename (Module.obj_name m) ~ext:Melange.js_ext
-  in
+  let name = Melange.make_module_name m ^ Melange.js_ext in
```

Comment:
Should this call `Path.extend_basename`?

Reply:
I am not sure. To use `extend_basename` here I would need to convert the return type from `make_module_name` to `Path.t`, but it's not really a path, it's just the name of the source file without any extension. 🤔

Reply:
Fair. the implementation is effectively the same. `basename ^ ext`. Seems fine.